server/src/main/java/org/elasticsearch/index/codec/vectors/ES813Int8FlatVectorFormat.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.codec.vectors;
+
+import org.apache.lucene.codecs.FlatVectorsFormat;
+import org.apache.lucene.codecs.FlatVectorsReader;
+import org.apache.lucene.codecs.FlatVectorsWriter;
+import org.apache.lucene.codecs.KnnFieldVectorsWriter;
+import org.apache.lucene.codecs.KnnVectorsFormat;
+import org.apache.lucene.codecs.KnnVectorsReader;
+import org.apache.lucene.codecs.KnnVectorsWriter;
+import org.apache.lucene.codecs.lucene99.Lucene99ScalarQuantizedVectorsFormat;
+import org.apache.lucene.index.ByteVectorValues;
+import org.apache.lucene.index.FieldInfo;
+import org.apache.lucene.index.FloatVectorValues;
+import org.apache.lucene.index.MergeState;
+import org.apache.lucene.index.SegmentReadState;
+import org.apache.lucene.index.SegmentWriteState;
+import org.apache.lucene.index.Sorter;
+import org.apache.lucene.search.KnnCollector;
+import org.apache.lucene.util.Bits;
+import org.apache.lucene.util.hnsw.OrdinalTranslatedKnnCollector;
+import org.apache.lucene.util.hnsw.RandomVectorScorer;
+
+import java.io.IOException;
+
+public class ES813Int8FlatVectorFormat extends KnnVectorsFormat {
+
+    static final String NAME = "ES813Int8FlatVectorFormat";
+
+    private final FlatVectorsFormat format;
+
+    public ES813Int8FlatVectorFormat() {
+        this(null);
+    }
+
+    /**
+     * Sole constructor
+     */
+    public ES813Int8FlatVectorFormat(Float confidenceInterval) {
+        super(NAME);
+        this.format = new Lucene99ScalarQuantizedVectorsFormat(confidenceInterval);
```

Comment:
If we get a new Lucene format for vectors, will we just this class here? Or will we create anew ES<V>Int9FlatVectorFormat?

Reply:
@mayya-sharipova it depends on backwards compatibility. I think we would end up with an `ES814Int8FlatVectorFormat.java` with the new versions and the `ES813Int8FlatVectorFormat.java` would become read only (SPI should create teh correct thing to load it up) and `ES813Int8FlatVectorFormat` will have to use a backwards format `Lucene99ScalarQuantizedVectorsFormat` or reader. 

HNSW is in the same boat as it uses these things.

Reply:
@jpountz any thoughts here? Have we don't anything like this similarly in the past?

Reply:
This is my expectation as well.